node/full_node_test.go

```diff
@@ -25,78 +25,102 @@ import (
 	"github.com/rollkit/rollkit/test/mocks"
 )
 
-// simply check that node is starting and stopping without panicking
+// TestStartup checks if the node starts and stops without any errors
 func TestStartup(t *testing.T) {
 	assert := assert.New(t)
 	require := require.New(t)
 
-	app := &mocks.Application{}
-	app.On(InitChain, mock.Anything).Return(abci.ResponseInitChain{})
-	key, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	signingKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	node, err := newFullNode(ctx, config.NodeConfig{DALayer: "mock"}, key, signingKey, proxy.NewLocalClientCreator(app), &types.GenesisDoc{ChainID: "test"}, log.TestingLogger())
-	require.NoError(err)
-	require.NotNil(node)
-
-	assert.False(node.IsRunning())
-
-	err = node.Start()
-	assert.NoError(err)
-	defer func() {
-		assert.NoError(node.Stop())
-	}()
-	assert.True(node.IsRunning())
+	ctx := context.Background()
+	node := initializeAndStartNode(ctx, require, assert)
+	defer cleanUpNode(node, assert)
 }
 
 func TestMempoolDirectly(t *testing.T) {
 	assert := assert.New(t)
 	require := require.New(t)
 
+	ctx := context.Background()
+	node := initializeAndStartNode(ctx, require, assert)
+	defer cleanUpNode(node, assert)
+
+	peerID := getPeerID(assert)
+	verifyTransactions(node, peerID, assert)
+	verifyMempoolSize(node, assert)
+}
+
+// cleanUpNode stops the node and checks if it is running
+func cleanUpNode(node *FullNode, assert *assert.Assertions) {
+	defer node.cancel()
+	assert.NoError(node.Stop())
+	assert.False(node.IsRunning())
+}
+
+// setupMockApplication initializes a mock application
+func setupMockApplication() *mocks.Application {
 	app := &mocks.Application{}
 	app.On(InitChain, mock.Anything).Return(abci.ResponseInitChain{})
 	app.On(CheckTx, mock.Anything).Return(abci.ResponseCheckTx{})
+	return app
+}
+
+// generateSingleKey generates a single private key
+func generateSingleKey() crypto.PrivKey {
 	key, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	signingKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	anotherKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
+	return key
+}
+
+// newTestNode creates a new test node
+func newTestNode(ctx context.Context, key crypto.PrivKey, signingKey crypto.PrivKey, app *mocks.Application) (*FullNode, error) {
+	return newFullNode(ctx, config.NodeConfig{DALayer: "mock"}, key, signingKey, proxy.NewLocalClientCreator(app), &types.GenesisDoc{ChainID: "test"}, log.TestingLogger())
```

Comment:
do you need to use the default 1s blocktime and 15s da times or you can make it smaller to reduce the test run time?

Reply:
The current file tests run in ~0.542s on my machine, don't think any further optimizations benefit us at this point.